backend/app/crud/llm.py

```
@@ -0,0 +1,228 @@
+        conversation_id=conversation_id,
+        auto_create=auto_create,
+        config=config_dict,
+    )
+
+    session.add(db_llm_call)
+    session.commit()
+    session.refresh(db_llm_call)
+
+    logger.info(
+        f"[create_llm_call] Created LLM call id={db_llm_call.id}, "
+        f"job_id={job_id}, provider={provider}, model={model}"
```

Ruff (0.14.14)

[error] Undefined name `provider`

(F821)